src/shared/options/readOptions.test.ts

```diff
@@ -0,0 +1,86 @@
+import { describe, expect, it, vi } from "vitest";
+
+import { readOptions } from "./readOptions.js";
+
+const emptyOptions = {
+	author: undefined,
+	base: undefined,
+	createRepository: undefined,
+	description: undefined,
+	email: undefined,
+	excludeCompliance: undefined,
+	excludeContributors: undefined,
+	excludeLintJson: undefined,
+	excludeLintKnip: undefined,
+	excludeLintMd: undefined,
+	excludeLintPackageJson: undefined,
+	excludeLintPackages: undefined,
+	excludeLintPerfectionist: undefined,
+	excludeLintSpelling: undefined,
+	excludeLintYml: undefined,
+	excludeReleases: undefined,
+	excludeRenovate: undefined,
+	excludeTests: undefined,
+	funding: undefined,
+	owner: undefined,
+	repository: undefined,
+	skipGitHubApi: false,
+	skipInstall: false,
+	skipRemoval: false,
+	skipRestore: undefined,
+	skipUninstall: false,
+	title: undefined,
+};
+
+const mockOptions = {
+	base: "prompt",
+	github: "mock.git",
+	repository: "mock.repository",
+};
+
+vi.mock("./getPrefillOrPromptedOption.js", () => ({
+	get getPrefillOrPromptedOption() {
+		return vi.fn(() => "mock");
+	},
+}));
+
+vi.mock("./ensureRepositoryExists.js", () => ({
+	get ensureRepositoryExists() {
+		return vi.fn(() => ({
+			github: mockOptions.github,
+			repository: mockOptions.repository,
+		}));
+	},
+}));
+
+vi.mock("../../shared/cli/spinners.ts", () => ({
+	get withSpinner() {
+		return vi.fn(() => ({}));
+	},
+}));
+
+vi.mock("./augmentOptionsWithExcludes.js", () => ({
+	get augmentOptionsWithExcludes() {
+		return vi.fn(() => ({ ...emptyOptions, ...mockOptions }));
+	},
+}));
+
+describe("readOptions", () => {
+	it("validates unsupported argument and cancels the function", async () => {
```

Comment:
[Naming] This is super nitpicky and I won't mind if you angrily shut this down 😄 but: I generally try to name tests as _X when Y_. I find it more clear for conveying what the _cause_ (_Y_) is and the _effect_ (_X_). Thoughts?

```suggestion
	it("cancels the function when --base is invalid", async () => {
```

(here and the other one too)

Reply:
Yep makes sense